pcp/versions/2.0/inventory.md

```diff
@@ -11,32 +11,68 @@ A client node that wants an inventory list of a PCP broker must 1) be
 connected to it (refer to [sessions][1]) and 2) send an inventory
 request message to a PCP broker.
 
-In case the request is processed successfully, the broker will reply with an
-inventory response message.
+If the request is processed successfully, the broker will reply with an
+inventory response message. Additionally, if the inventory request includes a
+`subscribe_to_updates` flag set to `true`, the broker will send the client an
+`inventory update` message when the broker's inventory changes. This implies
+that the broker must track two registries of clients: those connected, in
+addition to those that have subscribed to updates.
 
 Inventory Messages
 ---
 
-Inventory request and response messages must have the *message_type*
-field respectively equal to `http://puppetlabs.com/inventory_request` and
-`http://puppetlabs.com/inventory_response`.
-
-The *in_reply_to* field of an inventory response contains the ID of the
-inventory request.
-
-The *data* field of an inventory request message is an array of URIs,
-described by the following json-schema:
+PCP specifies three three types of inventory messages: inventory requests,
+inventory responses, and inventory updates.
 
+For inventory requests:
+* The *message_type* field must equal `http://puppetlabs.com/inventory_request` 
+* The *data* field must be an array of URIs conforming to the following
+  json-schema:
 ```
 {
     "type" : "array",
     "items" : { "type" : "string",
                 "pattern" : "^pcp://[^/]*/[^/]+$" }
 }
 ```
+* When provided, the *subscribe_to_updates* flag must be set to `true` or
+  `false`.
+
+For inventory responses:
+* The *message_type* field must equal `http://puppetlabs.com/inventory_response`
+* The *in_reply_to* field must contain the ID of the inventory request.
+* The *data* field must be an array of URIs as described for inventory
+  requests.
+
+For inventory updates:
+* The *message_type* field must equal `http://puppetlabs.com/inventory_update`
+* The *data* field must be a JSON map conforming to the following json-schema:
+  ```
+{
+  "properties": {
+    "version": { "type": "string",
+                 "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[0-9a-f]{4}-[0-9a-f]{12}_\d+$" },
+    "changes": {"type": "array",
+                "items": {
+                  "properties": {
+                    "client": { "type": "string",
+                                "pattern": "^pcp://[^/]*/[^/]+$" },
+                    "change": { "type": "integer",
+                                "description": "A representation of a connection or disconnection. 1 for connection, -1 for disconnection"}
+                  }
+                }
+    }
+  }
+}
+```
+
+In reference to the schema above, the `version` field is the concatenation of a
+UUID4, an underscore, and an integer. The UUID is generated by the broker at
```

Comment:
What is the UUID supposed to identify?
The particular inventory update message or rather the broker that sent it?